### PR TITLE
10 Add (optional) server/api with address checking and notifications

### PR DESCRIPTION
**Pull request by paOol**
replaced rest, so it's important to be thorough to Q/A for issues.

### CODE REVIEW

File: react/src/hooks/useAddressDetails.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { useEffect, useState } from 'react';
 import { AddressDetails, getAddressDetails } from '../util/api-client';
 import { validateCashAddress } from '../util/address';
 
-const POLL_DELAY = 100;
+const POLL_DELAY = 1000;
```
Comment (Klakurka) [inline react/src/hooks/useAddressDetails.ts:6]: What's `POLL_DELAY` for?

File: react/src/util/api-client.ts [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+    `https://api.paybutton.org/address/transactions/${address}`,
   );
   return await res.json();
 };
 
+export const getSatoshiBalance = async (
+  address: string,
+): Promise<{ satoshis: number }> => {
+  const res = await fetch(
+    `https://api.paybutton.org/address/balance/${address}`,
```
Comment (Klakurka) [inline react/src/util/api-client.ts:25]: Can we move `https://api.paybutton.org/` to a parameter where the default value is `https://api.paybutton.org/` ?

That way it's all in 1 place and can be changed easily by people who decide to run their own server.

File: react/src/util/api-client.ts [hunk 3]
```diff
@@ -1,14 +1,37 @@
...
+  );
+  return await res.json();
+};
+
+export const getUTXOs = async (address: string): Promise<UtxoDetails> => {
+  const res = await fetch(`https://api.paybutton.org/address/utxo/${address}`);
```
Comment (Klakurka) [inline react/src/util/api-client.ts:31]: Same here.

File: react/src/util/api-client.ts [hunk 4]
```diff
@@ -19,11 +42,20 @@ export const getFiatPrice = async (currency: currency): Promise<PriceData> => {
...
+//   );
+//   return await res.json();
+// };
+
 export const getTransactionDetails = async (
   txid: string,
 ): Promise<TransactionDetails> => {
   const res = await fetch(
-    `https://rest.bitcoin.com/v2/transaction/details/${txid}`,
+    `https://api.paybutton.org/transactions/details/${txid}`,
```
Comment (Klakurka) [inline react/src/util/api-client.ts:58]: And here.

**Review by Klakurka — CHANGES_REQUESTED** [review]

File: react/src/util/api-client.ts [hunk 5]
```diff
@@ -20,15 +20,17 @@ export const getAddressDetails = async (
 
 export const getSatoshiBalance = async (
   address: string,
+  rootUrl = `https://api.paybutton.org/`,
```
Comment (Klakurka) [inline react/src/util/api-client.ts:23]: What do you think about moving that last `/` into the `fetch()` calls? Do you think that's better or worse?

Comment (gnarj) [reply react/src/util/api-client.ts:23]: I think that's a much better way to do it. I'll get that changed asap.

File: react/src/util/api-client.ts [hunk 6]
```diff
@@ -1,14 +1,39 @@
...
+//   return await res.json();
+// };
+
 export const getAddressDetails = async (
   address: string,
 ): Promise<AddressDetails> => {
   const res = await fetch(
-    `https://rest.bitcoin.com/v2/address/details/${address}`,
+    `https://api.paybutton.org/address/transactions/${address}`,
   );
```
Comment (gnarj) [inline react/src/util/api-client.ts:17]: @Klakurka Should we change this url to be a parameter as well?

Comment (Klakurka) [reply react/src/util/api-client.ts:17]: Ya anything that uses the api should be I think.

**Comment by Klakurka** [comment]
Code looks good, just need to test it.

**Comment by Klakurka** [comment]
I might be testing incorrectly but I tried running the normal JS version using rollup and it's still pointing to rest.bitcoin.com.

1. cd paybutton && npm i
2. npm start
3. Site will be running on localhost:10001.

**Review by Klakurka — CHANGES_REQUESTED** [review]
Rollup way isn't updating to use api.paybutton.org for me for some reason.

**Comment by gnarj** [comment]
I was able to solve this by running:
cd paybutton/react
npm run build

**Comment by Klakurka** [comment]
I get this at build time:

image

**Pull request merged**